Formula/openssl110.rb

```diff
@@ -0,0 +1,184 @@
+    no-ssl3
+    no-ssl3-method
+    no-zlib
+    shared
+    enable-cms
+  ]
+  end
+
+  def install
+    if build.universal?
+      ENV.permit_arch_flags
+      archs = Hardware::CPU.universal_archs
+    elsif MacOS.prefer_64_bit?
+      archs = [Hardware::CPU.arch_64_bit]
+    else
+      archs = [Hardware::CPU.arch_32_bit]
+    end
+
+    dirs = []
+
+    archs.each do |arch|
+      if build.universal?
+        dir = "build-#{arch}"
+        dirs << dir
+        mkdir dir
+        mkdir "#{dir}/engines"
+      end
+
+      ENV.deparallelize
+      system "perl", "./Configure", *(configure_args + arch_args[arch])
+      system "make", "clean" if build.universal?
+      system "make", "depend"
+      system "make"
+      system "make", "test" if build.with?("test")
+
+      if build.universal?
+        cp "include/openssl/opensslconf.h", dir
+        cp Dir["*.?.?.dylib", "*.a", "apps/openssl"], dir
+        cp Dir["engines/**/*.dylib"], "#{dir}/engines"
+      end
+    end
+
+    system "make", "install", "MANDIR=#{man}", "MANSUFFIX=ssl"
+
+    if build.universal?
+      %w[libcrypto libssl].each do |libname|
+        system "lipo", "-create", "#{dirs.first}/#{libname}.1.1.dylib",
+                                  "#{dirs.last}/#{libname}.1.1.dylib",
+                       "-output", "#{lib}/#{libname}.1.1.dylib"
+        system "lipo", "-create", "#{dirs.first}/#{libname}.a",
+                                  "#{dirs.last}/#{libname}.a",
+                       "-output", "#{lib}/#{libname}.a"
+      end
+
+      Dir.glob("#{dirs.first}/engines/*.dylib") do |engine|
+        libname = File.basename(engine)
+        system "lipo", "-create", "#{dirs.first}/engines/#{libname}",
+                                  "#{dirs.last}/engines/#{libname}",
+                       "-output", "#{lib}/engines/#{libname}"
+      end
+
+      system "lipo", "-create", "#{dirs.first}/openssl",
+                                "#{dirs.last}/openssl",
+                     "-output", "#{bin}/openssl"
+
+      confs = archs.map do |arch|
+        <<-EOS.undent
+          #ifdef __#{arch}__
+          #{(buildpath/"build-#{arch}/opensslconf.h").read}
+          #endif
+          EOS
+      end
+      (include/"openssl/opensslconf.h").atomic_write confs.join("\n")
+    end
+  end
+
+  def openssldir
+    etc/"openssl110"
+  end
+
+  def post_install
+    # OpenSSL "remembers" the Perl it was built against. Where that is
+    # Homebrew's Perl it remembers the Cellar path, which breaks every version.
+    # Fix this manually to use the opt path which remains consistent.
+    perl_scripts = %W[
+      #{openssldir}/misc/CA.pl
+      #{openssldir}/misc/tsget
+      #{bin}/c_rehash
+    ]
+    perl_scripts.each do |f|
+      next unless File.exist?(f)
+      if File.read(f).include?(Formula["perl"].bin)
+        inreplace f, Formula["perl"].bin, Formula["perl"].opt_bin
+      end
+    end
+
+    keychains = %w[
+      /Library/Keychains/System.keychain
```

Comment:
And I've been sitting on this one for _a while actually_ because, honestly ,I didn't know what would be an appropriately-private forum. So this is kinda **burn-after-reading**. If there's a better place to talk about this, let me know.

But essentially, this line is not such a good idea.
As in:

Say I have a self-signed root CA and a leaf SSL cert in the System keychain (not uncommon). In Apple-land, these get big ❌s, warnings, etc. But the `.pem` file we're dumping is a just _blind whitelist_.

Example:
1. go to https://geoff.codes
You should get a big invalid certificate modal dialog.

2. `wget https://geoff.codes`
You should also get stopped here because of certificate problems.

3. `cat [1.pem.txt](https://github.com/Homebrew/homebrew-core/files/254015/1.pem.txt) [2.pem.txt](https://github.com/Homebrew/homebrew-core/files/254016/2.pem.txt)  >> "$(brew --prefix)/etc/openssl/cert.pem"

4. `wget https://geoff.codes`
😨

(Github made me add .txt to the pem files to upload, but if you rename them back to .pem, Spotlight will quickly show you that the emperor has no clothes.)